development/build/scripts.js

```diff
@@ -333,6 +334,7 @@ function createScriptTasks ({ browserPlatforms, livereload }) {
           ? '00000000000000000000000000000000'
           : undefined
       ),
+      SEGMENT_WRITE_KEY: opts.testing ? undefined : conf.SEGMENT_WRITE_KEY,
```

Comment:
We should probably use a separate key for production builds v.s. non-production (i.e. PR or release candidate) builds.

Maybe one for prod, one for everything else? :thinking: Or three, with one one additional key for RCs or any test builds that we distribute.

Or is there some other way to separate test metrics from production? Sentry has the `environment` property for example.

Reply:
There's a development key that we could use for PRs, which I think would be fine. The purpose of the development source in Segment is to debug events, and ensure that they are tracked to plan. I'm curious if we could override SEGMENT_WRITE_KEY in CI for release candidate builds?

Would the RC builds have a special version number returned from ```platform.getVersion()``` or no?

Reply:
I don't know of a way to change the CI environment variable in different scenarios. There's just one set of variables for the whole project. Using two separate keys seems easiest. We can still collapse them to `SEGMENT_WRITE_KEY` when we inject it with `envify`.

The only way to identify the RC builds would be the `environment` variable (injected as `METAMASK_ENVIRONMENT`). That should say whether it's a release candidate, a PR, a "staging" build (i.e. `develop`), or something else (e.g. a local build made with `yarn dist`).

Reply:
Just using the dev key for anything non-production seems OK to me for now though! Adding the third key was just an idea. We can revisit it if we think that would be useful.

Reply:
handled in https://github.com/MetaMask/metamask-extension/pull/9382/commits/af67e9661b14ae050cda6a65e8d8d845225bc03f

Reply:
This now includes network, platform, environment, environment_type and version in context instead of properties.

Reply:
well shit, I guess this doesn't actually solve the problem of where the events land -- I do think PRs should hit development. RCs probably should go to prod? I'll ask Lorenzo.

Reply:
I would suggest that prod only be "real users", on official builds. Anyone using a different build is likely a developer or a tester. Until we create a separate dashboard for RCs, I think dev would make sense. The dev dashboard is likely to be low-volume enough to find relevant metrics from the RCs anyway.

Reply:
that works for me

Reply:
The `context` solution looks great! That certainly simplifies things.

Reply:
https://github.com/MetaMask/metamask-extension/pull/9382/commits/e912c0b0c9892219a130d5b3681c5e48fb4799a3 final piece of the puzzle, i think